### DISCUSSION TITLE
Webhooks Payload send as a POST request

### DISCUSSION THREAD

**Discussion by Krish-farmd** [Budibase/budibase#14338] [Help] [answered]
Can Budibase send a Webhook POST request to send a row of data as a payload with the conditional trigger?
I got the webhooks only for getting the data so far.
Can anyone please confirm or help me to do the same?

**Comment by melohagan** [comment, marked as answer]
Hey @Krish-farmd 

I believe you are looking for the [External Data Connector](https://docs.budibase.com/docs/external-data-connector) automation step.

This will work with a POST [REST query](https://docs.budibase.com/docs/rest-queries)

**Reply by jnprovence1** [reply]
I've read the documentation multiple times now - I want to do the same thing as OP, however, my post request just goes to a post API call, it doesn't actually return results.  The use case is taking the data to get embedded in a vector store.

**Reply by melohagan** [reply]
If you need to return data from an automation, you will want to use the **Collect Data** step: https://docs.budibase.com/docs/synchronous-automations#creating-a-synchronous-automation

**Reply by jnprovence1** [reply]
I don't need to return data.  I'm just trying to take a value from a (created/updated/deleted) row and send it to an external API on a different server.  There will be no data to return.

**Reply by melohagan** [reply]
There's triggers for Row Created, Row Updated and Row Deleted for the internal Budibase DB.

**Reply by jnprovence1** [reply]
yep, I've got the triggers with the bindings and I've created the endpoints on the other server, I just can't figure out how to tell budibase to send the data to the api endpoint.  Even if I create an external data connection, I don't see where you can pass a variable in for the POST body.

**Reply by melohagan** [reply]
Could you share some screenshots of your automation please?

Ah there may be a recent related bug: https://github.com/Budibase/budibase/issues/14531

**Reply by jnprovence1** [reply]
Here is my REST API screen (not sure what to put in for the body section and automation screen.  Thank you for taking a look

![image](https://github.com/user-attachments/assets/13216318-e1ea-40f7-95da-c0b3d273e4f6)


![image](https://github.com/user-attachments/assets/42757433-50b9-4ea6-9fb0-a8387eeaff31)

**Reply by melohagan** [reply]
So in the `Bindings` tab I assume you have a binding named **id**.

Your post body should then be: 

```js
{ 
  "Id": {{ id }} 
}
```

**Reply by jnprovence1** [reply]
It seems to want to use {{ Binding.id }}, but either way, here is what I get.

![image](https://github.com/user-attachments/assets/31688d54-0f48-48ff-a731-6fd2545c3a38)
![image](https://github.com/user-attachments/assets/221dfcbd-1722-489d-8bfb-234f5e71c908)

**Reply by melohagan** [reply]
You could try using `{{ trigger.row.id }}` 

**Note the lowercase id**
